### DISCUSSION TITLE
Building problem

### DISCUSSION THREAD

**Discussion by halentech** [complexlogic/flex-launcher#55] [Q&A]
Hello Iam trying to build X86 version from sources and it is not sucessfull. I install latest visual studio 2022 with latest required components and failed with last two steps:  cmake --build OR cmake --build . --config Release --target package 

lot of SDL errors. I tried X64 version too, but same errors.
Please could someone help? I need X86 version

[error.txt](https://github.com/complexlogic/flex-launcher/files/14786840/error.txt)

**Comment by complexlogic** [comment]
I don't officially support 32 bit, but it should build fine.

Linker errors typically indicate some type of binary incompatibility. If you're using the CMake command verbatim from the compilation guide, that could be causing an issue because it builds 64 bit versions of the dependencies, including SDL.

Delete your entire build folder and start over from new. Try this for the CMake command:
```
cmake .. -G "Visual Studio 17 2022" -A Win32 -DCMAKE_TOOLCHAIN_FILE=".\vcpkg\scripts\buildsystems\vcpkg.cmake" -DVCPKG_TARGET_TRIPLET="x86-windows-static"
```

That should build 32 bit versions of both the dependencies and the program, eliminating any linker errors.

**Reply by halentech** [reply]
Hello complexlogic,
I tried it again. Deleted build folder and used those commands:

git clone https://github.com/microsoft/vcpkg
and
cmake .. -G "Visual Studio 17 2022" -A Win32 -DCMAKE_TOOLCHAIN_FILE=".\vcpkg\scripts\buildsystems\vcpkg.cmake" -DVCPKG_TARGET_TRIPLET="x86-windows-static"

and result is same. Error count is: 111 SDL2 errors
It seems, that there is some problem with link to SDL2 libraries 

log file from building in the english lang...

[errorx86.txt](https://github.com/complexlogic/flex-launcher/files/14837350/errorx86.txt)

**Reply by complexlogic** [reply]
After some investigation, these build errors were due to a recent change in the vcpkg toolchain. Had nothing to do with 64 or 32 bit configuration.

I updated the build script so everything is working now. You will have to pull the latest commit from `master`.

**Reply by halentech** [reply]
Hello complexlogic,

x64 build success, x86 no. I got error:

platform.lib(win32.obj) : error LNK2019: unresolved external symbol _SetSuspendState referenced in function _scmd_sleep

I found, that maybe is missing **powrprof.lib** as referenced library.

https://stackoverflow.com/questions/15851692/how-to-use-setsuspendstate-api-for-hibernate-or-suspend-in-win32-sdk

[error2.txt](https://github.com/complexlogic/flex-launcher/files/15013562/error2.txt)
